### PR TITLE
feat: refactor all component to ember v4

### PR DESCRIPTION
**Pull request by anehx**
BREAKING CHANGE: Please check [the migration guide](/docs/migration-v5) for instructions on how to update to the new version. This version drops support for Ember < v3.24 and requires `ember-auto-import` v2.

### CODE REVIEW

**Comment by anehx** [comment]
@velrest @czosel @luytena I realize that this PR is way too big to properly review. It's basically a complete rewrite of the addon in Ember v4. However, you can verify that nothing (except what's stated in the [migration guide](https://github.com/anehx/ember-uikit/blob/updates/docs/migration-v5.md)) changed if you check the tests which were only refactored slightly.

You might just need to trust me that I didn't break everything... :wink:

**Comment by anehx** [comment]
Closes #733

File: addon/components/uk-button.hbs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  class={{trimmed-join
+    "uk-button"
+    (if @active "uk-active")
+    this.color
+    this.size
+    this.width
+  }}
+  disabled={{@disabled}}
+  title={{@title}}
+  {{on "click" (optional @onClick)}}
```
Comment (czosel) [inline addon/components/uk-button.hbs:12]: is it still possible to trigger the action by focussing the button and then clicking `Enter`? This works with the previous version, even though I'm not sure why :see_no_evil:

Comment (anehx) [reply addon/components/uk-button.hbs:12]: Yes this works, this is plain accessibility in browsers.

**Review by czosel — APPROVED** [review]
:100: read every line

**Pull request merged**